Build failure: ezquake

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix-build -A ezquake`, nixpkgs commit `cad22e7d996aea55ecab064e84834289143e44a0`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

[x86_64-linux](https://hydra.nixos.org/build/317990294), [aarch64-linux](https://hydra.nixos.org/build/317990291)

### Relevant log output

```console
error: Cannot build '/nix/store/c167mvg13p94x5qfl2scgp2jkng7qmw6-ezquake-3.6.3.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/mqs7ra9vgmg3lvlzsynh2ij251maz4iw-ezquake-3.6.3
       Last 25 log lines:
       >    39 | typedef enum {false, true} qbool;
       >       |               ^~~~~
       > src/q_shared.h:39:15: note: 'false' is a keyword with '-std=c23' onwards
       > In file included from src/common.h:35,
       >                  from src/crc.c:25:
       > src/q_shared.h:39:15: error: cannot use keyword 'false' as enumeration constant
       >    39 | typedef enum {false, true} qbool;
       >       |               ^~~~~
       > src/q_shared.h:39:15: note: 'false' is a keyword with '-std=c23' onwards
       > In file included from src/common.h:35,
       >                  from src/common.c:23:
       > src/q_shared.h:39:15: error: cannot use keyword 'false' as enumeration constant
       >    39 | typedef enum {false, true} qbool;
       >       |               ^~~~~
       > src/q_shared.h:39:15: note: 'false' is a keyword with '-std=c23' onwards
       > make: *** [Makefile:637: .ezquake/src/crc.o] Error 1
       > make: *** Waiting for unfinished jobs....
       > make: *** [Makefile:637: .ezquake/src/cmodel.o] Error 1
       > make: *** [Makefile:637: .ezquake/src/cmd.o] Error 1
       > make: *** [Makefile:637: .ezquake/src/com_msg.o] Error 1
       > src/common.c: In function 'COM_WriteToUniqueTempFile':
       > src/common.c:429:17: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
       >   429 |                 write(fd, buffer, bytes);
       >       |                 ^~~~~~~~~~~~~~~~~~~~~~~~
       > make: *** [Makefile:637: .ezquake/src/common.o] Error 1
       For full logs, run:
         nix log /nix/store/c167mvg13p94x5qfl2scgp2jkng7qmw6-ezquake-3.6.3.drv
```

### Additional context

_No response_

### System metadata

```
 - system: `"aarch64-linux"`
 - host os: `Linux 6.17.12, NixOS, 26.05 (Yarara), 26.05.20251215.1306659`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.32.4`
 - nixpkgs: `/nix/store/nv0f6mcdixfwhjqxnnrk3iwfvkgbrxjl-source`
```

### Notify maintainers

@edwtjo 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ez/ezquake/package.nix
+++ b/pkgs/by-name/ez/ezquake/package.nix
@@ -7,46 +7,45 @@
   jansson,
   libpng,
   libjpeg,
-  libGLU,
   libGL,
   libX11,
   libsndfile,
-  libXxf86vm,
-  pcre,
+  pcre2,
+  minizip,
+  cmake,
   pkg-config,
   SDL2,
-  vim,
-  speex,
 }:
 
 stdenv.mkDerivation rec {
   pname = "ezquake";
-  version = "3.6.3";
+  version = "3.6.8";
 
   src = fetchFromGitHub {
     owner = "QW-Group";
     repo = "ezquake" + "-source";
     tag = version;
     fetchSubmodules = true;
-    hash = "sha256-ThrsJfj+eP7Lv2ZSNLO6/b98VHrL6/rhwf2p0qMvTF8=";
+    hash = "sha256-BIkBl6ncwo0NljuqOHJ3yQeDTcClh5FGssdFsKUjN90=";
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
   buildInputs = [
+    minizip
+    pcre2
     expat
     curl
     jansson
     libpng
     libjpeg
-    libGLU
     libGL
     libsndfile
     libX11
-    libXxf86vm
-    pcre
     SDL2
-    vim
-    speex
   ];
 
   installPhase =

--- a/pkgs/by-name/ez/ezquake/package.nix
+++ b/pkgs/by-name/ez/ezquake/package.nix
@@ -1,30 +1,31 @@
 {
-  lib,
-  stdenv,
-  fetchFromGitHub,
-  curl,
+  SDL2,
+  cmake,
+  curlMinimal,
   expat,
+  fetchFromGitHub,
   jansson,
-  libpng,
-  libjpeg,
+  lib,
   libGL,
   libX11,
+  libjpeg,
+  libpng,
   libsndfile,
-  pcre2,
   minizip,
-  cmake,
+  nix-update-script,
+  pcre2,
   pkg-config,
-  SDL2,
+  stdenv,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "ezquake";
   version = "3.6.8";
 
   src = fetchFromGitHub {
     owner = "QW-Group";
-    repo = "ezquake" + "-source";
-    tag = version;
+    repo = "ezquake-source";
+    tag = finalAttrs.version;
     fetchSubmodules = true;
     hash = "sha256-BIkBl6ncwo0NljuqOHJ3yQeDTcClh5FGssdFsKUjN90=";
   };
@@ -35,18 +36,20 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
+    SDL2
+    curlMinimal
+    expat
+    jansson
+    libGL
+    libX11
+    libjpeg
+    libpng
+    libsndfile
     minizip
     pcre2
-    expat
-    curl
-    jansson
-    libpng
-    libjpeg
-    libGL
-    libsndfile
-    libX11
-    SDL2
   ];
+
+  enableParallelBuilding = true;
 
   installPhase =
     let
@@ -54,12 +57,15 @@ stdenv.mkDerivation rec {
       arch = lib.head (lib.splitString "-" stdenv.hostPlatform.system);
     in
     ''
-      mkdir -p $out/bin
-      find .
-      mv ezquake-${sys}-${arch} $out/bin/ezquake
+      runHook preInstall
+
+      install -D \
+        ezquake-${sys}-${arch} $out/bin/${finalAttrs.meta.mainProgram}
+
+      runHook postInstall
     '';
 
-  enableParallelBuilding = true;
+  passthru.updateScript = nix-update-script { };
 
   meta = {
     homepage = "https://ezquake.com/";
@@ -69,4 +75,4 @@ stdenv.mkDerivation rec {
     platforms = lib.platforms.linux;
     maintainers = with lib.maintainers; [ edwtjo ];
   };
-}
+})

